Remove Travis build shield from README

mokagio opened the pull request:
We don't use Travis anymore, since #1091.

### Test
N.A

### Review
Only one developer required to review these changes, but anyone can perform the review.

### Release
These changes do not require release notes.

mokagio committed:
Remove Travis build shield from README

We don't use Travis anymore, since #1091.

mokagio commented on README.md:
```diff
@@ -1,5 +1,5 @@
 # Simplenote for Android
-[![Version](https://img.shields.io/badge/version-2.19-blue.svg)](https://github.com/Automattic/simplenote-android/releases/tag/2.19) [![CircleCI](https://img.shields.io/circleci/build/gh/Automattic/simplenote-android.svg?label=circleci)](https://circleci.com/gh/Automattic/simplenote-android) [![Travis CI](https://img.shields.io/travis/Automattic/simplenote-android/develop.svg?label=travisci)](htt ...
+[![Version](https://img.shields.io/badge/version-2.19-blue.svg)](https://github.com/Automattic/simplenote-android/releases/tag/2.19) [![CircleCI](https://img.shields.io/circleci/build/gh/Automattic/simplenote-android.svg?label=circleci)](https://circleci.com/gh/Automattic/simplenote-android)
```
For what is worth, I'd say we should remove the version and the CircleCI shields, too. Those shields provide information that is already available on GitHub:

<img width="1252" alt="Screen Shot 2021-08-23 at 1 06 44 pm" src="https://user-images.githubusercontent.com/1218433/130384619-83d2d9a7-1d2a-48b9-8914-1d6de59a3a5d.png">

AliSoftware replied:
I agree, let's remove them all.

This information (CI status and latest version) was not as visible a while ago with previous GitHub UI, but now that it is, it's not really relevant anymore.

(In which case we also need to remove the step that bumps that version in the badge during release finalization btw)

mokagio committed:
Remove version and CircleCI build status shields from README

GitHub already shows that information in the UI.

See discussion at:
https://github.com/Automattic/simplenote-android/pull/1455#discussion_r693718418

AliSoftware reviewed (approved):
@mokagio How was the `Version` badge – that this PR removed – updated on every release before? Since that's a step we might want to remove as well

I've tried a quick search of the term `README` in the Fastfile and scripts and couldn't find anything that seem to automate the update of that badge during release steps, and couldn't find anything in the release scenario either to suggest to do that step manually…

So was this all just manual and implicit and never documented in the scenario, which means nothing to remove on that front either? (Approving as I'm assuming 👈 this is the case)

mokagio commented:
> that's a step we might want to remove as well

100%. In fact, I did remove it while working on this, but never opened a PR in our private repo with the release process checklist 😅
